packages/ibm-products/src/components/Coachmark/Coachmark.js

```diff
@@ -0,0 +1,265 @@
+/**
+ * Copyright IBM Corp. 2023, 2023
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { forwardRef, useEffect, useRef, useState } from 'react';
+import cx from 'classnames';
+import PropTypes, { Component } from 'prop-types';
+import { createPortal } from 'react-dom';
+import { CoachmarkOverlay } from './CoachmarkOverlay';
+import { CoachmarkContext } from './utils/context';
+import { COACHMARK_OVERLAY_KIND } from './utils/enums';
+import { useClickOutsideElement, useWindowEvent } from './utils/hooks';
+import { getDevtoolsProps } from '../../global/js/utils/devtools';
+import { pkg /*, carbon */ } from '../../settings';
+import { throttle } from 'lodash';
+// The block part of our conventional BEM class names (blockClass__E--M).
+const blockClass = `${pkg.prefix}--coachmark`;
+const componentName = 'Coachmark';
+
+const defaults = {
+  align: 'bottom',
+  onClose: () => {},
+  overlayKind: 'tooltip',
+  theme: 'light',
+};
+
+/**
+ * Coachmarks are used to call out specific functionality or concepts
+ * within the UI that may not be intuitive but are important for the
+ * user to gain understanding of the product's main value and discover new use cases.
+ */
+
+export let Coachmark = forwardRef(
+  (
+    {
+      align = defaults.align,
+      children,
+      className,
+      onClose = defaults.onClose,
+      overlayClassName,
+      overlayKind = defaults.overlayKind,
+      overlayRef,
+      portalSelector,
```

Comment:
We have a `portalTarget` prop in the Tearsheet component. Might be a good idea to keep the same naming convention, instead of `portalSelection`.